### PR TITLE
Rails initializer class loading

### PR DESCRIPTION
**Pull request by thilton-oddball**
## Description of change
This PR gets rid of the deprecation warnings (that would eventually become errors) related to autoloading constants during initialization. By wrapping code in initializer files in 'Rails.application.reloader.to_prepare do' blocks, it eliminates the warning and allows any changes made to classes to get reloaded in the development environment.

Reloading does not reboot the application, and therefore code executed during initialization does not run again. So, if you reload, changes to classes in the initializer won't be reflected in that stale Class object because the initializer only runs on reboot. Therefore, code that references a reloadable class or module, but that is not executed again on reload, becomes stale.

Co-authored by: @LindseySaari

### CODE REVIEW

**Comment by LindseySaari** [comment]
@thilton-oddball I think this may be ready to go, but I wasn't 100% sure on the rubocop changes

File: .rubocop.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ Style/ZeroLengthPredicate:
   Enabled: false
 
 Layout/LineLength:
-  Max: 120
+  Max: 125
```
Comment (ericboehs) [inline .rubocop.yml:90]: 80 and 120 are a pretty standard line lengths. Why the change to 125?

Comment (thilton-oddball) [reply .rubocop.yml:90]: I can change it back to 120, I mainly just tweaked it because one of the lines that had to change ended up being like 122 and Rubocop was giving me grief.

File: modules/mobile/config/initializers/statsd.rb [hunk 2]
```diff
@@ -88,17 +100,6 @@
 StatsD.increment('mobile.profile.link_account.success', 0)
 StatsD.increment('mobile.profile.link_account.failure', 0)
```
Comment (ericboehs) [inline modules/mobile/config/initializers/statsd.rb:101]: Having a hard time figuring out why this file has so many changes. It looks the same. 🤔

Comment (thilton-oddball) [reply modules/mobile/config/initializers/statsd.rb:101]: I was taking lines out of the `Rails.application.reloader.to_prepare do` block and then putting them back in trying to figure out what needed to be in there and what didn't, so I probably ended up getting things out of the order they were originally in.

Comment (ericboehs) [reply modules/mobile/config/initializers/statsd.rb:101]: Why not just put them all in there?

Comment (thilton-oddball) [reply modules/mobile/config/initializers/statsd.rb:101]: We certainly could I guess. I'll make that change and push it up.

File: .rubocop.yml [hunk 3]
```diff
@@ -129,6 +129,7 @@ Metrics/BlockLength:
     - 'modules/*/spec/**/*.rb'
     - 'modules/*/app/swagger/**/*.rb'
     - 'modules/*/config/routes.rb'
+    - 'modules/mobile/config/initializers/statsd.rb'
```
Comment (ericboehs) [inline .rubocop.yml:132]: Can you add this to the `.rubocop_todo.yml` instead via `rubocop --auto-gen-config`? You may have to run `rubocop -a` afterwards to correct any redundant disables.

**Comment by ericboehs** [comment]
You forgot the `rubocop -a` 😺

**Comment by thilton-oddball** [comment]
https://gfycat.com/bouncycarefuldormouse-patrick-stewart-star-trek-facepalm-picard

**Review by ericboehs — APPROVED** [review]
